mando/core.py

```diff
@@ -119,7 +142,16 @@ def parse(self, args):
         then be called as ``command(*args)``.
 
         :param args: The arguments to parse.'''
-        arg_map = self.parser.parse_args(args).__dict__
+        try:
+            # run completion handler before parsing
+            import argcomplete
+            argcomplete.autocomplete(self.parser)
+        except ImportError:  # pragma: no cover
```

Comment:
I looked into `argcomplete` and it looks good. Keeping it as an optional dependency seems like the correct choice here. I'm opening a new issue to remember to mention it in the documentation.